controllers/common/common.go

```diff
@@ -245,6 +257,33 @@ func FullNameForPolicy(plc *policiesv1.Policy) string {
 	return plc.GetNamespace() + "." + plc.GetName()
 }
 
+// GetRepPoliciesInPlacementBinding returns a list of the replicated policies that are either direct subjects of
+// the given PlacementBinding, or are in PolicySets that are subjects of the PlacementBinding.
+// The list items are guaranteed to be unique (for example if a policy is in multiple sets).
+func GetRepPoliciesInPlacementBinding(
+	ctx context.Context, c client.Client, pb *policiesv1.PlacementBinding,
+) []reconcile.Request {
+	decisions, err := GetDecisions(ctx, c, pb)
+	if err != nil {
+		return []reconcile.Request{}
+	}
+	// Use this for removing duplicated policies
+	rootPolicyRequest := GetPoliciesInPlacementBinding(ctx, c, pb)
+
+	result := []reconcile.Request{}
```

Comment:
To reduce memory allocations.
```suggestion
	result := make([]reconcile.Request, 0, len(rootPolicyRequest) * len(decisions))
```